### DISCUSSION TITLE
Issue when modifying a M2M field

### DISCUSSION THREAD

**Discussion by arladmin** [directus/directus#7325] [Archive]
_Directus v9.0.0-rc.89_

Hi all. I've come across an issue when modifying a M2M field.

In my create (parent) item (POST) API call, i'm unable to set any value for the M2M field.
```Error: 403 - {"errors":[{"message":"You don't have permission to access this.","extensions":{"code":"FORBIDDEN"}}]}```

This is the format that i'm implementing: https://docs.directus.io/reference/api/introduction/#one-to-many-many-to-many

Error is received in both cases:

Selecting an existing entry
```
POST call with body: {"M2M_field": [2] }  //2 = id of existing relational item
```
Creating a new entry
```
POST call with body: {"M2M_field": [{"a":"b"}] }  
```
The logs (set to debug mode) also don't show anything other than the above mentioned error.
How to figure out what's the exact issue here?

**Edit:** Seems like a ```children``` property might be needed as well (based on the example in the docs).
So, now i tried this:
```
POST call with body: {"M2M_field": {"children": [2]} }
```
New error:
```
{
  "message": "Invalid one-to-many update structure: "children" is not allowed",
  "extensions": {
    "code": "INVALID_PAYLOAD"
  }
}
```

**Update:** 
I tried another thing-
```
POST call with body: {"M2M_field": [18] }  //  18 is the id of an existing 'M2M join table' item
```
This time the API call was a success.

- But why doesn't the API work when the information of the actual child item is provided, as given in the docs ?
- Mainly, what's the proper way to create/update an item while also creating/adding existing nested items, in the same API call?

**Comment by arladmin** [comment]
**Final update:**
Now able to successfully add nested data. Although, the way i've been able to achieve it is by using the M2M join table object, instead of the actual nested item object:
```
Body of POST call to create a Parent item-
{"M2M_field": [
  { "nested_field_name_in_M2M_join": { actual_nested_item_object } }   // {M2M_join_table_object}
//  or
  { "nested_field_name_in_M2M_join": id_of_actual_nested_item_object }   // {M2M_join_table_object}
] }
```

If this is indeed the intended method, it wasn't clear, at least to me, from the docs 😕

**Reply by rijkvanzanten** [reply]
A many to many is nothing more than a one-to-many followed by a many-to-one. Therefore, the regular rules and flows for updating a o2m apply to m2ms as well (https://docs.directus.io/reference/api/introduction/#relational-data).

In this case, yes, the `M2M_field` references the junction table, and the junction table has a field references a further related collection.

**Reply by arladmin** [reply]
Yeah, finally figured that out after several hit and trials. Even though the structure of an M2M relation was clear, didn't find anywhere in the docs regarding the usage of 'junction table data' in the actual API calls.

Maybe my non-dev background contributed to this lack of understanding.

**Reply by rijkvanzanten** [reply]
(cc @jacobrienstra  / @alesvaupotic)

**Reply by benhaynes** [reply]
So excited to setup GitHub teams so we can `@docs`! 👍
